src/core/version_neg.c

```diff
@@ -396,101 +273,85 @@ QuicVersionNegotiationExtEncodeVersionNegotiationInfo(
         //
         // Generate Server VNI.
         //
-        VNILen = sizeof(uint32_t) + QuicVarIntSize(DesiredVersionsListLength) +
-            (DesiredVersionsListLength * sizeof(uint32_t));
+        VILen = sizeof(uint32_t) + (DesiredVersionsListLength * sizeof(uint32_t));
 
-        VersionNegotiationInfo = CXPLAT_ALLOC_NONPAGED(VNILen, QUIC_POOL_VER_NEG_INFO);
-        if (VersionNegotiationInfo == NULL) {
+        VersionInfo = CXPLAT_ALLOC_NONPAGED(VILen, QUIC_POOL_VERSION_INFO);
+        if (VersionInfo == NULL) {
             QuicTraceEvent(
                 AllocFailure,
                 "Allocation of '%s' failed. (%llu bytes)",
-                "Server Version Negotiation Info",
-                VNILen);
+                "Server Version Info",
+                VILen);
             return NULL;
         }
-        VNIBuf = VersionNegotiationInfo;
+        VIBuf = VersionInfo;
 
-        CxPlatCopyMemory(VNIBuf, &Connection->Stats.QuicVersion, sizeof(Connection->Stats.QuicVersion));
-        VNIBuf += sizeof(Connection->Stats.QuicVersion);
-        VNIBuf = QuicVarIntEncode(DesiredVersionsListLength, VNIBuf);
+        _Analysis_assume_(VILen >= sizeof(uint32_t));
```

Comment:
Instead of assumes, use debug asserts. They also assume, but will crash in debug mode if invalid. But, assuming you got a build error because of this, the only reason I can think of this failing is if there is wrap around because of the addition and multiplication above. It might be worth adding an assert **there** instead to ensure wrap around would be impossible.